homeassistant/components/blebox/__init__.py

```diff
@@ -65,25 +68,21 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 
 @callback
-def create_blebox_entities(
-    hass, config_entry, async_add_entities, entity_klass, entity_type
-):
-    """Create entities from a BleBox product's features."""
-
-    product = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
-    entities = []
+def get_blebox_features(
+    hass: HomeAssistant, config_entry: ConfigEntry, entity_type: str
+) -> list[Feature]:
+    """Get features from a BleBox product matching the feature type."""
 
+    product: Box = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
     if entity_type in product.features:
-        for feature in product.features[entity_type]:
-            entities.append(entity_klass(feature))
-
-    async_add_entities(entities, True)
+        return product.features[entity_type]
+    return []
```

Comment:
Why not use `get`?
```py
return product.features.get(entity_type, [])
```

Reply:
Implemented - but moved to each platform